lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1326,6 +1365,26 @@ DIFile *SPIRVToLLVMDbgTran::getFile(const SPIRVId SourceId) {
   return getDIFile(getString(SourceArgs[FileIdx]), ParseChecksum(ChecksumStr));
 }
 
+DIBuilder &SPIRVToLLVMDbgTran::getDIBuilder(const SPIRVExtInst *DebugInst) {
+  assert(BuilderMap.size() != 0 && "No debug compile units");
+  if (BuilderMap.size() == 1)
+    return *BuilderMap.begin()->second;
+  while (DebugInst->getExtOp() != SPIRVDebug::CompilationUnit) {
+    if (DebugInst->getExtOp() == SPIRVDebug::DebugInfoNone)
+      return *BuilderMap.begin()->second;
+    const SPIRVWordVec &Ops = DebugInst->getArguments();
+    SPIRVWord ParentScopeIdx = 0;
+    if (!hasDbgInstParentScopeIdx(DebugInst->getExtOp(), ParentScopeIdx))
+      return *BuilderMap.begin()->second;
```

Comment:
Just to understand, do we always take first Builder by default because Instructions with no scope are related to the "default" Builder?

Reply:
Builder here is being used to create dependencies between scope lowering them to CU scope. If we have no scope information - we translate such debug info with the best effort possible. If scope is just File - we actually don't really care, what builder to use. If CU is linked with a particular file, then debugger can link Debug info with the appropriate CU with the same File scope.